### PR TITLE
Convert all int to int32/64 in preparation for protobuf

### PR DESCRIPTION
**Pull request by smarterclayton**
Aligns with upstream.  Does not contain sdn changes yet.

[test]

### CODE REVIEW

**Comment by mfojtik** [comment]
@smarterclayton I thought I already catched all int<->int32 conversions in the rebase

**Comment by smarterclayton** [comment]
No.  Internal types need to be intXX as well, and UodatePercent,
LatestVersion, and some network ones were not correct.

On Jun 16, 2016, at 4:35 AM, Michal Fojtik <notifications@github.com> wrote:

@smarterclayton <https://github.com/smarterclayton> I thought I already
catched all int<->int32 conversions in the rebase

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/9366#issuecomment-226422589>, or mute
the thread
<https://github.com/notifications/unsubscribe/ABG_pwCTJZGXvcUdmfa62hVwzhiMQATjks5qMQrUgaJpZM4I2_i3>
.

**Comment by smarterclayton** [comment]
@kargakis this is going to break all your PRs - just making you aware.

**Comment by smarterclayton** [comment]
@mfojtik if you can review I'd appreciate it

File: pkg/image/api/types.go [hunk 1]
```diff
@@ -74,8 +74,8 @@ type Image struct {
 type ImageLayer struct {
 	// Name of the layer as defined by the underlying store.
 	Name string
-	// Size of the layer as defined by the underlying store.
-	Size int64
+	// LayerSize of the layer as defined by the underlying store.
+	LayerSize int64
```
Comment (mfojtik) [inline pkg/image/api/types.go]: what is the different between these two? can you update the godoc?

File: pkg/image/api/v1/types.go [hunk 2]
```diff
@@ -40,8 +40,8 @@ type Image struct {
 type ImageLayer struct {
 	// Name of the layer as defined by the underlying store.
 	Name string `json:"name"`
-	// Size of the layer as defined by the underlying store.
-	Size int64 `json:"size"`
+	// Size of the layer in bytes as defined by the underlying store.
```
Comment (mfojtik) [inline pkg/image/api/v1/types.go]: ok, can you unify the comments?

File: pkg/deploy/api/v1beta3/conversion_test.go [hunk 3]
```diff
@@ -206,7 +206,7 @@ func newInt64(val int64) *int64 {
 	return &val
 }
 
-func newInt(val int) *int {
+func newInt32(val int32) *int32 {
```
Comment (mfojtik) [inline pkg/deploy/api/v1beta3/conversion_test.go]: `newInt32Ptr` ? I don't think this is returning a new int32, just a pointer to existing one.

**Comment by mfojtik** [comment]
@smarterclayton LGTM

File: pkg/deploy/api/v1beta3/conversion_test.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/deploy/api/v1beta3/conversion_test.go]: Consistent with all the other test code.

File: pkg/image/api/v1/types.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/image/api/v1/types.go]: Can't, this is public API doc so Godoc should match JSON.

Comment (mfojtik) [inline pkg/image/api/v1/types.go]: gotcha.

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[test]

**Pull request merged**